src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -264,76 +261,61 @@ public static LinePositionSpan GetLinePositionSpan(DiagnosticDataLocation? dataL
         {
             var lines = text.Lines;
             if (lines.Count == 0)
-            {
                 return default;
-            }
-
-            var dataLocationStartLine = (useMapped ? dataLocation?.MappedStartLine : dataLocation?.OriginalStartLine) ?? 0;
-            var dataLocationStartColumn = (useMapped ? dataLocation?.MappedStartColumn : dataLocation?.OriginalStartColumn) ?? 0;
-            var dataLocationEndLine = (useMapped ? dataLocation?.MappedEndLine : dataLocation?.OriginalEndLine) ?? 0;
-            var dataLocationEndColumn = (useMapped ? dataLocation?.MappedEndColumn : dataLocation?.OriginalEndColumn) ?? 0;
-
-            if (dataLocationStartLine >= lines.Count)
-            {
-                var lastLine = lines.GetLinePosition(text.Length);
-                return new LinePositionSpan(lastLine, lastLine);
-            }
-
-            AdjustBoundaries(dataLocationStartLine, dataLocationStartColumn, dataLocationEndLine, dataLocationEndColumn, lines,
-                out var startLine, out var startColumn, out var endLine, out var endColumn);
-
-            var startLinePosition = new LinePosition(startLine, startColumn);
-            var endLinePosition = new LinePosition(endLine, endColumn);
-            SwapIfNeeded(ref startLinePosition, ref endLinePosition);
 
-            return new LinePositionSpan(startLinePosition, endLinePosition);
-        }
-
-        public static TextSpan GetTextSpan(DiagnosticDataLocation? dataLocation, SourceText text)
-        {
-            var linePositionSpan = GetLinePositionSpan(dataLocation, text, useMapped: false);
-
-            var span = text.Lines.GetTextSpan(linePositionSpan);
-            return EnsureInBounds(TextSpan.FromBounds(Math.Max(span.Start, 0), Math.Max(span.End, 0)), text);
-        }
+            var startLine = (useMapped ? dataLocation?.MappedStartLine : dataLocation?.OriginalStartLine) ?? 0;
+            var endLine = (useMapped ? dataLocation?.MappedEndLine : dataLocation?.OriginalEndLine) ?? 0;
 
-        private static void AdjustBoundaries(int dataLocationStartLine, int dataLocationStartColumn, int dataLocationEndLine, int dataLocationEndColumn,
-            TextLineCollection lines, out int startLine, out int startColumn, out int endLine, out int endColumn)
-        {
-            startLine = dataLocationStartLine;
-            var originalStartColumn = dataLocationStartColumn;
+            // Make sure the starting columns are never negative.
+            var startColumn = Math.Max((useMapped ? dataLocation?.MappedStartColumn : dataLocation?.OriginalStartColumn) ?? 0, 0);
+            var endColumn = Math.Max((useMapped ? dataLocation?.MappedEndColumn : dataLocation?.OriginalEndColumn) ?? 0, 0);
 
-            startColumn = Math.Max(originalStartColumn, 0);
             if (startLine < 0)
             {
+                // If the start line is negative (e.g. before the start of the actual document) then move the start to the 0,0 position.
```

Comment:
once we're fully on pull we should be able to remove these adjustments right?  As we'd always be asking for diagnostics as of the snapshot we're on, so we wouldn't have stale ones

Reply:
correct.  :)

Reply:
i believe some of this is also because roslyn is involved with Build diagnsotics (pre LSP).  And we cannot guarantee that hte build diagnostics have any sort of invariants around them, or know what document snapshot versions they are associated with.